packages/cli/src/render-ts/renderRecsV1Tables.ts

```diff
@@ -18,14 +18,20 @@ export function defineContractComponents(world: World) {
 
 function renderDefineComponent(table: RecsV1TableOptions["tables"][number]) {
   const { namespace, name } = table.staticResourceData;
+  const tableId = new TableId(namespace, name);
   return `
     (() => {
-      const tableId = new TableId("${namespace}", "${name}");
       return defineComponent(world, {
-        ${table.fields.map(({ name, recsTypeString }) => `${name}: ${recsTypeString},`).join("")}
+        ${table.fields.map(({ name, recsTypeString }) => `${name}: ${recsTypeString}`).join(",")}
       }, {
-        metadata: { contractId: tableId.toHex(), tableId: tableId.toString() },
-      });
+        id: ${JSON.stringify(tableId.toHex())},
+        metadata: {
+          componentName: ${JSON.stringify(name)},
+          tableName: ${JSON.stringify([namespace, name].join(":"))},
```

Comment:
why do we need to run `JSON.stringify` on these? Aren't they already strings?

Reply:
this is generating JS code and namespace/name are technically user generated, so it's a bit safer to encode to string with `JSON.stringify` rather than wrapping it in quotes